ledger/acctupdates.go

```diff
@@ -842,7 +843,10 @@ func (aul *accountUpdatesLedgerEvaluator) LookupAsset(rnd basics.Round, addr bas
 }
 
 func (aul *accountUpdatesLedgerEvaluator) LookupKv(rnd basics.Round, key string) (*string, error) {
-	panic("not implemented")
+	// TODO: should I explicitly be avoiding the lock as I'm doing here?
```

Comment:
Appreciate feedback here ^^^

Reply:
I doubt it.  How do the other lookup routines in `accountUpdatesLedgerEvaluator` work?  There should be ones for resources and accounts.

Reply:
The two [functions above](https://github.com/algorand/go-algorand/blob/0632f1f9ee599f5f865cb9ee0bf7a0fabc21e7d0/ledger/acctupdates.go#L835-L843) (`LookupApplication` and `LookupAsset`) don't take a lock. Similarly, neither does `GetCreatorForRound()` nor `LookupWithoutRewards()`.

The [2 year old comment on L779](https://github.com/algorand/go-algorand/blob/0632f1f9ee599f5f865cb9ee0bf7a0fabc21e7d0/ledger/acctupdates.go#L779) can be interpreted as implying that the purpose of `accountUpdatesLedgerEvaluator` is to allow not taking the lock.